README.md

```diff
@@ -1 +1,172 @@
-# rust-vmm-ci
\ No newline at end of file
+# rust-vmm-ci
+
+The `rust-vmm-ci` contains integration tests and [Buildkite]() pipeline
+definitions that are used for running the CI for all rust-vmm crates.
+
+Having a centralized place for the tests is one of the enabler for keeping the
+same quality standard for all crates in rust-vmm.
+
+## Getting Started with rust-vmm-ci
+
+To run the integration tests defined in the pipeline as part of the CI:
+
+1. Add rust-vmm-ci as a git submodule to your repository
+
+```bash
+git submodule add https://github/rust-vmm/rust-vmm-ci
+git add rust-vmm-ci/
+git commit -s -m "Added rust-vmm-ci as submodule"
+```
+
+2. Create the coverage test configuration file named coverage_config.json in
+the root of the repository. One example of the configuration file can be found
+in [coverage_config.json.sample](coverage_config.json.sample).
+
+The json must have the following fields:
+- *coverage_score*: The coverage of the repository.
+- *exclude_path*: This field is used for excluding files from the report. It
+  should be used to exclude autogenerated files. Files in *exclude_path* are
+  separated by one coma. If the repository does not have any autogenerated
+  files, *exclude_path* should be an empty string.
+- *crate_features*: `cargo kcov` does not build crate features by default. To
+  get the coverage report including optional features these need to be
+  specified in *crate_features* separated by coma. If the crate does not have
+  any features, this field should be empty.
+
+This file is required for the coverage integration so it needs to be added
+to the repository as well.
+
+3. Create a new pipeline definition in Buildkite. For this step ask one of the
+rust-vmm Buildkite [admins](CODEOWNERS) to create one for you. Add a pipeline
+step that is uploading the rust-vmm-ci pipeline:
+
+```bash
+buildkite-agent pipeline upload rust-vmm-ci/.buildkite/pipeline.yml
+```
+
+4. The code owner of the repository will have to setup a WebHook for triggering
+the CI on Pull Requests and pushes.
+
+## Buildkite Pipeline
+
+The Buildkite pipeline is the definition of integration tests to be run as part
+of the CI. It includes steps for running unit tests and linters (including
+coding style checks), and computing the coverage.
+
```

Comment:
This section is a little confusing to me. It feels like there's a need to define that pipeline whereas in reality this is what the git submodule brings.
So I would refer back to step 1 of the previous section, explaining that this steps adds a buildkite pipeline into the repo and that buildkite will use it.

Reply:
This section is not about using rust-vmm-ci, but about how we created rust-vmm-ci and what does it have. It's a "developer" readme. Should I put it in a different .md file and keep the readme only with information about how to use rust-vmm-ci?

Reply:
I did some rephrasing here and there. After some more thinking I decided it's better to have only one readme for now. Maybe we should have both sections (Buildkite Pipeline and Integration Tests) in another section, but I can't find a good name. Maybe "rust-vmm-ci development"? What do you say?

Reply:
+1 for one single readme.
This looks fine to me.